src/spyglass/spikesorting/v1/metric_curation.py

```diff
@@ -0,0 +1,605 @@
+    # Spike sorting and parameters for metric curation. Use `insert_selection` to insert a row into this table.
+    metric_curation_id: varchar(32)
+    ---
+    -> CurationV1
+    -> WaveformParameters
+    -> MetricParameters
+    -> MetricCurationParameters
+    """
+
+    @classmethod
+    def insert_selection(cls, key: dict):
+        """Insert a row into MetricCurationSelection with an
+        automatically generated unique metric curation ID as the sole primary key.
+
+        Parameters
+        ----------
+        key : dict
+            primary key of CurationV1, WaveformParameters, MetricParameters MetricCurationParameters
+
+        Returns
+        -------
+        key : dict
+            key for the inserted row
+        """
+        if len((cls & key).fetch()) > 0:
+            print(
+                "This row has already been inserted into MetricCurationSelection."
+            )
+            return (cls & key).fetch1()
+        key["metric_curation_id"] = generate_nwb_uuid(
+            (SpikeSortingSelection & key).fetch1("nwb_file_name"),
+            "MC",
+            6,
+        )
+        cls.insert1(key, skip_duplicates=True)
+        return key
+
+
+@schema
+class MetricCuration(dj.Computed):
+    definition = """
+    # Results of applying curation based on quality metrics. To do additional curation, insert another row in `CurationV1`
+    -> MetricCurationSelection
+    ---
+    -> AnalysisNwbfile
+    object_id: varchar(40) # Object ID for the metrics in NWB file
+    """
+
+    def make(self, key):
+        # FETCH
+        nwb_file_name = (
+            SpikeSortingSelection * MetricCurationSelection & key
+        ).fetch1("nwb_file_name")
+
+        waveform_params = (
+            WaveformParameters * MetricCurationSelection & key
+        ).fetch1("waveform_params")
+        metric_params = (
+            MetricParameters * MetricCurationSelection & key
+        ).fetch1("metric_params")
+        label_params, merge_params = (
+            MetricCurationParameters * MetricCurationSelection & key
+        ).fetch1("label_params", "merge_params")
+        sorting_id, curation_id = (MetricCurationSelection & key).fetch1(
+            "sorting_id", "curation_id"
+        )
+        # DO
+        # load recording and sorting
+        recording = CurationV1.get_recording(
+            {"sorting_id": sorting_id, "curation_id": curation_id}
+        )
+        sorting = CurationV1.get_sorting(
+            {"sorting_id": sorting_id, "curation_id": curation_id}
+        )
+        # extract waveforms
+        if "whiten" in waveform_params:
+            if waveform_params.pop("whiten"):
+                recording = sp.whiten(recording, dtype=np.float64)
+
+        waveforms_dir = (
+            os.environ.get("SPYGLASS_TEMP_DIR")
+            + "/"
+            + key["metric_curation_id"]
+        )
+        try:
+            os.mkdir(waveforms_dir)
+        except FileExistsError:
+            pass
```

Comment:
the mkdir `-p` flag will ignore errors if already exists. `settings.py` already ensures that `temp_dir` exists

```suggestion
        (
            nwb_file_name,
            waveform_params,
            metric_params,
            label_params,
            merge_params,
            sorting_id,
            curation_id,
        ) = (
            SpikeSortingSelection
            * WaveformParameters
            * MetricParameters
            * MetricCurationParameters
            * MetricCurationSelection
            & key
        ).fetch1(
            "nwb_file_name",
            "waveform_params",
            "metric_params",
            "label_params",
            "merge_params",
            "sorting_id",
            "curation_id",
        )

        # DO
        # load recording and sorting
        curation_key = {"sorting_id": sorting_id, "curation_id": curation_id}
        recording = CurationV1.get_recording(curation_key)
        sorting = CurationV1.get_sorting(curation_key)
        # extract waveforms
        if waveform_params.get("whiten"):
            recording = sp.whiten(recording, dtype=np.float64)

        waveforms_dir = temp_dir + "/" + key["metric_curation_id"]
```